Assets/ModScripts/Submodules/WhosWho.cs

```diff
@@ -0,0 +1,293 @@
+            return 7;
+        }
+        else if (Module.Bomb.GetOnIndicators().Count() == Module.Bomb.GetOffIndicators().Count())
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] The amount of lit and unlit indicators are equal, using the first LED.", ModuleID);
+            return 0;
+        }
+        else if (LEDcolors.Count(x => warmColors.Contains(x)) >= 3)
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Three or more LEDs are warm colors (red, orange, or yellow), using the seventh LED.", ModuleID);
+            return 6;
+        }
+        else if (Module.Bomb.GetPortPlateCount() <= Module.Bomb.GetPortCount())
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] The port plate count is less than or equal to the port count, using the fifth LED.", ModuleID);
+            return 4;
+        }
+        else if (SameColorTwiceInARow(LEDcolors))
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] Two matching LEDs are next to each other, using the second LED.", ModuleID);
+            return 1;
+        }
+        else if (Module.Bomb.GetPortPlates().Any(x => x.Contains("DVI") && (x.Contains("StereoRCA") || x.Contains("PS2"))))
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] A DVI-D port is on the same port plate as a Stereo RCA port or a PS/2 port, using the sixth LED.", ModuleID);
+            return 5;
+        }
+        else if ((Module.Bomb.GetBatteryCount(1) > 0 && Module.Bomb.GetBatteryCount(2) == 0) || (Module.Bomb.GetBatteryCount(2) > 0 && Module.Bomb.GetBatteryCount(1) == 0))
+        {
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] There are only D batteries or AA batteries, using the fourth LED.", ModuleID);
+            return 3;
+        }
+        else
+        {
+            string[] ordinals = new string[]{"first", "second", "third", "fourth", "fifth", "sixth", "seventh", "eighth"};
+            UnityEngine.Debug.LogFormat("[The Cruel Modkit #{0}] None of the conditions applied, using the {1} LED.", ModuleID, ordinals[Module.Bomb.GetSerialNumberNumbers().Sum() % 8]);
+            return Module.Bomb.GetSerialNumberNumbers().Sum() % 8;
+        }
+    }
+
+    private bool SameColorTwiceInARow(string[] colors)
+    {
+        for(int i = 0; i < colors.Length - 1; i++)
+            if (colors[i] == colors[i+1])
+                return true;
+        return false;
+    }
+
+    public override void OnBulbButtonPress(int Button)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.Bulbs[Button].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
```

Comment:
```suggestion
        if (!Module.IsSolving && !Module.CheckValidComponents())
```
Add this check so that it doesn't check components every time